annotations/upload_validation.py

```diff
@@ -0,0 +1,89 @@
+'''
+upload_validation.py
+author: aditi.gopalan
+Description: This script checks if manifests have been uploaded to Synapse 
+'''
+
+import synapseclient
+import pandas as pd
+import csv
+import argparse
+from datetime import datetime
+
+def get_folder_type_argument():
+    parser = argparse.ArgumentParser(description="Process Synapse folders and create a manifest.")
+    parser.add_argument("csv_file_path", help="Path to the input CSV file.")
+    parser.add_argument("folder_type", choices=["Publication", "Dataset", "Tool"],
+                        help="Specify folder type: 'Publication', 'Dataset', or 'Tool'")
+    args = parser.parse_args()
+    return args.csv_file_path, args.folder_type
+
+def get_folder_id_column(folder_type):
+    if folder_type == "Publication":
+        return "folderIdPublication"
+    elif folder_type == "Dataset":
+        return "folderIdDatasets"
+    elif folder_type == "Tool":
+        return "folderIdTools"
```

Comment:
Good suggestion!

Reply:
Since `folderIdPublication` is singular, we can do an if/else instead:

```
def get_folder_id_column(folder_type):
   if folder_type in ["Publication"]:  # in case we need to add more folder_types that use singular
      return "folderIdPublication"
   else:
      return f"folderId{folder_type}s"
```

:blush: